framework/src/base/MooseApp.C

```diff
@@ -406,6 +408,21 @@ MooseApp::MooseApp(InputParameters parameters)
     // Sleep to allow time for the debugger to attach
     std::this_thread::sleep_for(std::chrono::seconds(getParam<unsigned int>("stop_for_debugger")));
   }
+
+  // clone the master mesh if provided
+  if (parameters.isParamValid("_master_mesh"))
+  {
+    _action_warehouse.mesh() = parameters.get<const MooseMesh *>("_master_mesh")->safeClone();
+    // looks like safeClone is not cloning all
+    _action_warehouse.mesh()->prepare();
```

Comment:
I was hoping `safeClone` is enough but appears not. Maybe we should fix `safeClone`?

Reply:
`safeClone` works, we use it in SetupMeshAction.C